modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryModelListener.java

```diff
@@ -64,6 +64,22 @@ private void _validate(ObjectEntry objectEntry) {
 					"an-api-sort-must-be-related-to-an-api-endpoint");
 			}
 
+			if (Validator.isNull(
+					_objectEntryHelper.getObjectEntry(
+						objectEntry.getCompanyId(),
+						StringBundler.concat(
+							"id eq '", apiEndpointId,
+							"' and retrieveType eq 'collection'"),
+						"L_API_ENDPOINT"))) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					null,
+					"An API sort can only be associated to API endpoints " +
+						"with a \"collection\" retrieve type",
+					"an-api-sort-can-only-be-associated-to-api-endpoints-" +
+						"with-a-\"collection\"-retrieve-type");
```

Comment:
The value "collection" can be parametrized.

Reply:
Do you want me to parametrize it in the message too? I think that in the refactor we talked about yesterday I'll do it but if those changes aren't accepted I think it doesn't make sense.

Reply:
I think as long as we respect the casing that brian wants, we can do it.